ecs/src/lib.rs

```diff
@@ -0,0 +1,292 @@
+#![deny(warnings)]
+
+pub mod component;
+mod entity;
+mod world;
+
+pub use entity::{Entities, Entity};
+pub use world::World;
+
+#[cfg(test)]
+mod tests {
+    use std::{any, collections::HashSet};
+
+    use crate::component::{ComponentRegistry, Storage, StorageType};
+
+    use super::*;
+
+    #[test]
+    fn component_registry() {
```

Comment:
Maybe add some documentation on the tests to clear up what purpose they serve

Reply:
Most of them kind of test what they say they do.

Perhaps the issue is mostly that the tests are too big and generic and should be broken up in smaller, more specific, test?

Reply:
Yes, maybe it is just me who don't understand the ECS code fully so understanding their tests are more difficult